crates/shared/src/price_estimation/trade_finder.rs

```diff
@@ -164,7 +164,20 @@ impl TradeVerifier {
             },
         };
 
-        let return_data = self.simulator.simulate(call, overrides).await?;
+        let return_data = match self.simulator.simulate(call, overrides).await {
+            Ok(data) => data,
+            Err(SimulationError::Other(err)) => {
+                // In case we have a simulator error (network, service is down,
+                // etc.), we optimistically return the quote estimate without
+                // simulation. This is so we don't accidentally
+                tracing::warn!(?err, "trade simulation error");
```

Comment:
SHould we maybe put into the same warning message the call that was made. Then it might be easier to debug

Reply:
In theory, it should be included in the `err`. The issue is that not all simulator implementation bubble up the revert message. For example, Tenderly API just has a general "EVM reverted" error and doesn't include the revert message for non-verified contracts.